### ISSUE TITLE
ZIO Test: Assertion.forall causes MatchError

### ISSUE THREAD

**Issue by mitsutaka-takeda** [zio/zio#2668]
When I use Assertion.forall with non-List Iterable, it causes scala.MatchError because forall match the argument against List. Should we make more typesafe by making the signature of forall to Assertion[List[A]] or changing the implementation of forall to accept non-List Iterable?

```scala
import zio.test._
import zio.test.Assertion._

object failedToMatchWithForallAssertion extends DefaultRunnableSpec(
  suite("try to use forall with immutable.Vector"){
    test("immutable.Vector"){
      assert(scala.collection.immutable.Vector(4), forall(equalTo(4)))
    }
  }
)
```

**Comment by adamgfraser** [comment]
@mitsutaka-takeda This was fixed in #2282 and will be included in the next release candidate.

**Comment by mitsutaka-takeda** [comment]
Thanks! I missed the commit.
